### PR TITLE
Adding the documentation on partial download of custom artifacts

### PR DESCRIPTION
**Pull request by romil07**

### CODE REVIEW

File: docs/authoring/PartialArtifactsDownload.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+### Role of the above 2 data sources in this feature.
+
+
+#### artifacts
+This data source is expected to return the base level artifact items that the corresponding artifact has published. For instance, in the below picture, the artifact _rogoyal-test has published 3 artifacts i.e drop1, drop2, drop3. These 3 artifacts are listed here are the result of the "artifacts" data source.
+
+![artifacts](./images/artifacts.png)
+
+Note: All the artifacts are of type "folder" and hence are expand-able. This takes us to the next data source.
```
Comment (omeshp) [inline docs/authoring/PartialArtifactsDownload.md]: The artifacts can be of any type, the restriction is that this type needs to be mapped in browsableArtifactTypeMapping

File: docs/authoring/PartialArtifactsDownload.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+This doc describes what the artifact extension author needs to do to get this feature.
+
+1. Provide a data source with the name "artifacts".
+2. Provide a data source with the name "artifactItems".
+
+### Role of the above 2 data sources in this feature.
+
+
+#### artifacts
+This data source is expected to return the base level artifact items that the corresponding artifact has published. For instance, in the below picture, the artifact _rogoyal-test has published 3 artifacts i.e drop1, drop2, drop3. These 3 artifacts are listed here are the result of the "artifacts" data source.
```
Comment (omeshp) [inline docs/authoring/PartialArtifactsDownload.md]:
```suggestion
This data source is expected to return the list of artifacts that the artifact source has published. For instance, in the below picture, the artifact source _rogoyal-test has published 3 artifacts i.e drop1, drop2, drop3. These 3 artifacts listed here are the result of the "artifacts" data source.
```

File: docs/authoring/PartialArtifactsDownload.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+![artifacts](./images/artifacts.png)
+
+Note: All the artifacts are of type "folder" and hence are expand-able. This takes us to the next data source.
+
+#### artifactItems
+This data source returns the children items of a given artifact item. For example, in the below picture, drop1 artifact item has 2 children items i.e Application Tier and Tools.
```
Comment (omeshp) [inline docs/authoring/PartialArtifactsDownload.md]:
```suggestion
This data source returns the child items of a given artifact. For example, in the below picture, drop1 artifact has 2 child items i.e Application Tier and Tools.
```

File: docs/authoring/PartialArtifactsDownload.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+![artifactItems](./images/artifactItems.png)
+
+Note: Even the 2 children items here are also of type folder, expanding them is not supported. Right now we support only 2 levels of folder hierarchy.
+
+Since we need to know which data source to call for different types of artifact items, you need to specify the "artifactType" field in the "data" dictionary of the "artifacts" data source response. Corresponding to the artifactType, you need to specify the data source name of the artifact's children in the "browsableArtifactTypeMapping" dictionary. You can take a look at the team city extension [here](https://github.com/Microsoft/azure-pipelines-extensions/blob/a87e4580ac9fa2ded6155b2d8cc334948ba05c12/Extensions/TeamCity/Src/vss-extension.json#L300).
```
Comment (omeshp) [inline docs/authoring/PartialArtifactsDownload.md]: data source binding instead of data source name

**Review by omeshp — APPROVED** [review]

File: docs/authoring/PartialArtifactsDownload.md (hunk 1 above)
Comment (romil07) [reply docs/authoring/PartialArtifactsDownload.md]: This is just for example. I have mentioned how to use that through browsable mapping in a later step. But I will make a change here to be clearer.

**Pull request merged**